### ISSUE TITLE
Display nothing rather than a single blank line for binary files

### ISSUE THREAD

**Issue by bobsilverberg** [mozilla/addons-code-manager#909]
STR:

1. View any binary file in browse mode, e.g., https://code.addons-dev.allizom.org/en-US/browse/494431/versions/1532144/?path=META-INF%2Fcose.sig.

Expected result:

The file content area is empty.

Actual result:

There is a single blank line (labelled with a `1`) displayed in the content area.

Screenshot:

![Screenshot 2019-07-16 11 16 52](https://user-images.githubusercontent.com/142755/61306840-3ae01e00-a7bb-11e9-89eb-9474a0c70bf6.png)

**Comment by EnTeQuAk** [comment]
Server-side parts that can be tackled once this lands: https://github.com/mozilla/addons-server/issues/11782

**Comment by bobsilverberg** [comment]
We might not have to make any changes at all to support this. If https://github.com/mozilla/addons-server/issues/11782 lands, and the API then returns an empty string for `content`, I think it will all "just work". I will confirm this, but it seems like a simpler approach to do that.

@kumar303 What do you think?

**Comment by kumar303** [comment]
I'd like to see it in action but that sounds fine. We may need to display some kind of message like *binary content not shown* to avoid confusion but that could be done in a follow-up issue.

**Comment by bobsilverberg** [comment]
I did find one issue with the front end if the API just starts returning an empty string for content for binary files. The `CodeView` component still displays a single empty line of content, with the line number of `1`. 

I have traced this back to the fact that [`CodeView` uses `getLines`](https://github.com/mozilla/addons-code-manager/blob/59476bc01918d7073b5def2c36575907b9063025/src/components/CodeView/index.tsx#L88) to break the content up into multiple lines, and [`getLines` always returns an array with at least one entry](https://github.com/mozilla/addons-code-manager/blob/59476bc01918d7073b5def2c36575907b9063025/src/components/CodeView/utils.tsx#L37). So, we can fix this either by changing the code in `CodeView` to not even bother calling `getLines` if `content` is empty, or we could change `getLines` to return an empty array if `content` is empty. The latter seems like the better solution and also makes sense to me.

@kumar303 what do you think?

**Comment by kumar303** [comment]
> we could change `getLines` to return an empty array if `content` is empty. 

@bobsilverberg Yeah, that sounds right

**Comment by kumar303** [comment]
For future reference, I suggest following [the issue template](https://raw.githubusercontent.com/mozilla/addons-code-manager/master/.github/ISSUE_TEMPLATE.md). It really helps communicate the problem, which is helpful for those reviewing / QA'ing your patch.

**Comment by kumar303** [comment]
I find this issue description confusing. I think the STR are in https://github.com/mozilla/addons-code-manager/issues/896#issuecomment-507712312

**Comment by bobsilverberg** [comment]
> I find this issue description confusing. I think the STR are in [#896 (comment)](https://github.com/mozilla/addons-code-manager/issues/896#issuecomment-507712312)

Yes, I made a bit of a mess of the description of this issue by repurposing it. At this point the issue really is simply the blank line visible when viewing a binary file. We don't want to see that one blank line, but instead should see nothing. A follow-up should probably be filed to display a message about the file being binary instead.

**Comment by bobsilverberg** [comment]
@kumar303 I'm going to get rid of most of the description and make it simpler.

**Comment by AlexandraMoga** [comment]
@bobsilverberg I've looked at several binary files in the browse page and there is only a blank space in the code view displayed now, which is what was expected.
Examples:
https://code.addons-dev.allizom.org/en-US/browse/385621/versions/1655456?path=META-INF%2Fcose.sig
https://code.addons-dev.allizom.org/en-US/browse/390151/versions/1688198?path=popup%2Ffonts%2Fglyphicons-halflings-regular.woff

![image](https://user-images.githubusercontent.com/31961530/61369400-b19f1900-a898-11e9-858a-2df5f5dac008.png)


However, I've noticed in the labels that the compare view was also targeted, but I'm not seeing any changes there - i.e. binary files are still showing some content:

example:
https://code.addons-dev.allizom.org/en-US/compare/385621/versions/1655456...1688374/?path=META-INF%2Fcose.manifest
https://code.addons-dev.allizom.org/en-US/compare/385621/versions/1655456...1688374/?path=public%2Ffont%2FProximaNova-Sbold-webfont.woff

**Comment by bobsilverberg** [comment]
> 
> However, I've noticed in the labels that the compare view was also targeted, but I'm not seeing any changes there - i.e. binary files are still showing some content:
> 
> example:
> [code.addons-dev.allizom.org/en-US/compare/385621/versions/1655456...1688374/?path=META-INF%2Fcose.manifest](https://code.addons-dev.allizom.org/en-US/compare/385621/versions/1655456...1688374/?path=META-INF%2Fcose.manifest)

I can see that this file has content in the compare view. The file looks like a text file, but the Browse view does treat it like a binary file and displays no content, so there's an inconsistency here. This has to do with data coming back from the API, so I suggest an issue be filed on addons-server for this.

> [code.addons-dev.allizom.org/en-US/compare/385621/versions/1655456...1688374/?path=public%2Ffont%2FProximaNova-Sbold-webfont.woff](https://code.addons-dev.allizom.org/en-US/compare/385621/versions/1655456...1688374/?path=public%2Ffont%2FProximaNova-Sbold-webfont.woff)

The API is not returning any data for this file, so this does not appear to be an issue. The screen does display the diff summary (showing no changes), but that is expected at this point.

What leads you to suggest that file contents are being displayed for this file?

**Comment by AlexandraMoga** [comment]
> This has to do with data coming back from the API, so I suggest an issue be filed on addons-server for this.

I have filed https://github.com/mozilla/addons-server/issues/11856 for that



> What leads you to suggest that file contents are being displayed for this file?

I guess I've been misled by the diff summary displayed at the top, which made me think that there is a line in  there.
